### PR TITLE
Cocoapods 1.7.0 multiproject support (b/128794527)

### PR DESCRIPTION
**Pull request by maksymmalyhin**
- header imports fixes - use modular headers where required
- test project to validate different Cocoapods versions
- Travis configuration

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -209,6 +209,14 @@ jobs:
       script:
         - travis_retry ./scripts/if_changed.sh ./scripts/build.sh $PROJECT $PLATFORM $METHOD
 
+    # Validate all Cocoapods configurations
+    # This may take long time, so we would like to run it only once all other tests pass
+    - stage: cocoapods_compatibility_check
+      name: Validate all Cocoapods configurations 
+      script:
+        # TODO: ./scripts/if_changed.sh to ./scripts/if_cron.sh once testing finished
```
Comment (maksymmalyhin) [inline .travis.yml]: I would like to test it together with other tests before merging. I'll change it once it's tested.

File: CocoapodsIntegrationTest/CocoapodsIntegrationTest/AppDelegate.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//
+//  AppDelegate.h
```
Comment (paulb777) [inline CocoapodsIntegrationTest/CocoapodsIntegrationTest/AppDelegate.h]: update licenses and include in style.sh

File: CocoapodsIntegrationTest/scripts/build_with_environment.sh [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 Google
```
Comment (paulb777) [inline CocoapodsIntegrationTest/scripts/build_with_environment.sh]: 2019 here and other new files

File: .travis.yml [hunk 4]
```diff
@@ -209,6 +209,14 @@ jobs:
...
+        - travis_retry ./scripts/if_changed.sh ./CocoapodsIntegrationTest/scripts/build_all_environments.sh
```
Comment (paulb777) [inline .travis.yml]: Should it be if_cron now?

File: CocoapodsIntegrationTest/CocoapodsIntegrationTest/AppDelegate.m [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+@implementation AppDelegate
+
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  // Override point for customization after application launch.
+  return YES;
+}
+
+
+- (void)applicationWillResignActive:(UIApplication *)application {
```
Comment (ryanwilson) [inline CocoapodsIntegrationTest/CocoapodsIntegrationTest/AppDelegate.m]: Nit: if these unused methods are going to stay empty, I think we just just remove them from the file.

File: .travis.yml (hunk 4 above)
Comment (maksymmalyhin) [reply .travis.yml]: I would like to run it one more time from the PR. I'll change it to `if_cron` once the build passes.

**Comment by maksymmalyhin** [comment]
This build illustrates the checks to be run for cron builds https://travis-ci.org/firebase/firebase-ios-sdk/builds/517562628

File: GoogleUtilities/CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,6 @@
 # Unreleased
+- `GULAppDelegateSwizzler`: support of remote notification methods. (#2698)
```
Comment (paulb777) [inline GoogleUtilities/CHANGELOG.md]: Is this file intended to be in this PR?

Comment (maksymmalyhin) [reply GoogleUtilities/CHANGELOG.md]: Well, probably, the changelog changes should have been better be in a different PR, but I would prefer to merge the PR to avoid waiting for another build to finish.

**Comment by maksymmalyhin** [comment]
@paulb777 Thanks!
